spring-cloud-stream/src/main/java/org/springframework/cloud/stream/aggregate/AggregateApplicationBuilder.java

```diff
@@ -48,9 +50,12 @@ public SourceConfigurer from(Class<?> app) {
 	}
 
 	public void run(String[] parentArgs) {
-		ConfigurableApplicationContext parentContext = AggregateApplication.createParentContext(parentArgs);
-		SharedChannelRegistry sharedChannelRegistry = parentContext.getBean(SharedChannelRegistry.class);
-		List<AppConfigurer> apps = new ArrayList<AppConfigurer>();
+		ConfigurableApplicationContext parentContext = this.parentContext != null
+				? this.parentContext
+				: AggregateApplication.createParentContext(parentArgs);
+		SharedChannelRegistry sharedChannelRegistry = parentContext
```

Comment:
I think we handle the scenario where the `parentContext` (when set explicitly) doesn't have the `SharedChannelRegistry` bean.